overlord/devicestate/firstboot20_test.go

```diff
@@ -549,6 +561,85 @@ func (s *firstBoot20Suite) TestLoadDeviceSeedCore20(c *C) {
 	c.Assert(err, ErrorMatches, `internal error: requested inconsistent device seed: 20210201 \(was 20191018\)`)
 }
 
+func (s *firstBoot20Suite) setupSeed20DangerousGrade(c *C, withAutoImportAssertion bool) string {
+	m := boot.Modeenv{
+		Mode:           "run",
+		RecoverySystem: "20191018",
+		Base:           "core20_1.snap",
+	}
+
+	s.earlySetup(c, &m, "dangerous", "")
+
+	if withAutoImportAssertion {
+		seedtest.WriteValidAutoImportAssertion(c, s.Brands, s.SeedDir, m.RecoverySystem, 0644)
+	}
+
+	o, err := overlord.New(nil)
+	c.Assert(err, IsNil)
+	st := o.State()
+
+	st.Lock()
+	defer st.Unlock()
+
+	logbuf, restore := logger.MockLogger()
+	defer restore()
+
+	deviceSeed, err := devicestate.LoadDeviceSeed(st, m.RecoverySystem)
+	c.Assert(err, IsNil)
+
+	c.Check(deviceSeed.Model().BrandID(), Equals, "my-brand")
+	c.Check(deviceSeed.Model().Model(), Equals, "my-model")
+	c.Check(deviceSeed.Model().Base(), Equals, "core20")
+	c.Check(deviceSeed.Model().Grade(), Equals, asserts.ModelDangerous)
+
+	return logbuf.String()
+}
+
+func (s *firstBoot20Suite) TestLoadDeviceSeedCore20DangerousNoAutoImport(c *C) {
+	var calledcreateAllUsers = false
+	r := devicestate.MockCreateAllSystemUsers(func(state *state.State, assertDb asserts.RODatabase, model *asserts.Model, serial *asserts.Serial, sudoer bool) ([]*devicestate.CreatedUser, error) {
```

Comment:
The above suggestion applies to this test and the one below as well

Reply:
Also fixed as with the previous test case.